packages/esm-reports-app/src/components/reports-data-overview.component.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Select, SelectItem, DatePicker, DatePickerInput, Button } from '@carbon/react';
+import { ExtensionSlot, showSnackbar } from '@openmrs/esm-framework';
+import { useReportDefinitions, getReportDataObservable } from './reports.resource';
+import Overlay from './overlay.component';
+import ReportDataViewer from './report-data-viewer.component';
+import styles from './reports.scss';
+import { take } from 'rxjs/operators';
+import dayjs from 'dayjs';
+
+const ReportsDataOverviewComponent: React.FC = () => {
+  const { t } = useTranslation();
+  const [selectedReport, setSelectedReport] = useState('');
+  const [reportParameters, setReportParameters] = useState<Record<string, string>>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [reportData, setReportData] = useState(null);
+
+  const { reportDefinitions } = useReportDefinitions();
+
+  const selectedReportDefinition = useMemo(() => {
+    return reportDefinitions?.find((report) => report.uuid === selectedReport);
+  }, [selectedReport, reportDefinitions]);
+
+  const handleParameterChange = (parameterName: string, value: string) => {
+    const formattedDate = value ? dayjs(value).format('YYYY-MM-DDTHH:mm:ss.SSSZ') : '';
+    setReportParameters((prev) => ({
+      ...prev,
+      [parameterName]: formattedDate,
+    }));
+  };
+
+  const handleFetchReport = () => {
+    if (!selectedReport) {
+      showSnackbar({
+        kind: 'error',
+        title: t('error', 'Error'),
```

Comment:
For many really simple strings like this that are used in many places, we have a framework-provided `getCoreTranslation()` function that can be used like: `getCoreTranslation('error')`.